OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/InAppMessagesManager.kt

```diff
@@ -450,13 +450,14 @@ internal class InAppMessagesManager(
      *
      *
      * Make all messages with redisplay available if:
-     * - Already displayed
+     * - Already displayed in a different session
      * - At least one Trigger has changed
      */
     private fun makeRedisplayMessagesAvailableWithTriggers(newTriggersKeys: Collection<String>) {
         for (message in messages) {
             if (!message.isTriggerChanged && redisplayedInAppMessages.contains(message) &&
-                _triggerController.isTriggerOnMessage(message, newTriggersKeys)
+                _triggerController.isTriggerOnMessage(message, newTriggersKeys) &&
+                !message.isDisplayedInSession
```

Comment:
This breaks the "Every time trigger conditions are satisfied" logic so we shouldn't do this.

Reply:
Ok, I just pushed a different change that only target dynamic trigger firing logic.